boards/navigation/src/main.rs

```diff
@@ -0,0 +1,23 @@
+#![no_std]
+#![no_main]
+
+use embassy_executor::Spawner;
+use embassy_stm32::gpio::{Input, Pull};
+use embassy_time::{Duration, Timer};
+use hyped_io::gpio::EmbassyGpio;
+use hyped_sensors::keyence::Keyence;
+use {defmt_rtt as _, panic_probe as _};
+
+#[embassy_executor::main]
+async fn main(_spawner: Spawner) -> ! {
+    let p = embassy_stm32::init(Default::default());
+    let gpio_input = Input::new(p.PA0, Pull::Up);
+    let gpio = EmbassyGpio::new(gpio_input);
+    let mut keyence = Keyence::new(gpio);
+
+    loop {
```

Comment:
This isn't something we've clarified yet, but i think each sensor should have it's own task since the main task (which runs with the highest prio by default iirc) would be incharge of the can communication for that board (obviously this isn't something done yet but i think just implementing everything as it's own task is decent enough future proofing).

Reply:
This can ofc always be a ToDoLater for as soon as we figure that portion of the architecture out, up to you.

Reply:
Good point, will change :)

Reply:
Re no_std Vec stuff, I'm pretty sure we can't have std just for tests since they're co-located in the same file, but I could be wrong...

Reply:
Ah you're probably right, i think cargo flags can let us do something with it by just declaring std as an external crate in the test mode but that's ugly and probably not worth it 👍

Reply:
Ah okay, we could do that if we really need std for something in the future